include/silo/query_engine/operators/bitmap_producer.h

```diff
@@ -0,0 +1,41 @@
+#ifndef SILO_BITMAP_PRODUCER_H
+#define SILO_BITMAP_PRODUCER_H
+
+#include <cstdint>
+#include <functional>
+#include <memory>
+#include <string>
+
+#include "silo/query_engine/operator_result.h"
+#include "silo/query_engine/operators/operator.h"
+
+namespace roaring {
+class Roaring;
+}  // namespace roaring
+
+namespace silo::query_engine::operators {
+
+class BitmapProducer : public Operator {
```

Comment:
I thought, we can do without this function.
The plan in my head would have been to map the prefiltering and the regex match as seperate operators, which can then be reordered by the optimizer to further reduce intermediate result sizes.

But trying to refactor I noticed this was not easily possible, as the notion of prefiltering for **InsertionPositions**, not sequence ids can not be modelled by an operator tree.
Therefore I guess keeping this operator as is, is the way to go